### DISCUSSION TITLE
Is there an easy way to check if a cell is inside a LatLngPoly?

### DISCUSSION THREAD

**Discussion by luizamfsantos** [uber/h3#927] [Q&A] [answered]
I'm trying to match a list of cells to their corresponding polygon. My current approach: I converted the geojson to multilatlngpoly.
Then, I'm converting the LatLngPoly to h3cells in the same resolution as the cell I'm comparing to. Then I'm checking if the cell is in the list of cells. 
```{python}
geojson_df['latlngpoly'] = geojson_df['geometry'].apply(h3.geo_to_h3shape)
geojson_df['h3_cells_res7'] = geojson_df['latlngpoly'].apply(h3.h3shape_to_cells, res=7)
for idx_station, station in stations_df.iterrows():
        for idx_province, mpoly in geojson_df.iterrows():
            if station['h3_res7'] in mpoly[h3_cells_res7]:
                    stations_df.loc[idx_station,'province'] = mpoly['name']
                    continue
```
I was wondering if there's a better approach than what I'm doing.

**Comment by dfellis** [comment, marked as answer]
If you want to check containment of an individual cell within a polygon, just using [`cellToLatLng`](https://h3geo.org/docs/api/indexing#celltolatlng) and then using a normal point-in-polygon call with that point would be equivalent based on how [`polygonToCells`](https://h3geo.org/docs/api/regions#polygontocells) generates the set of cells returned.

If you want full containment or any overlapping containment (with the assumption that the polygon feature details are not smaller than an individual cell), you can use [`cellToBoundary`](https://h3geo.org/docs/api/indexing#celltoboundary) to get the borders of the cell in question, and then test all of those points with the point-in-polygon algorithm, where full containment requires all points to be in, and any overlap requires at least one.

If the set of polygons is relatively static and smaller in number than the set of cells you're testing, *then* converting the polygons into a list of cells, and converting that into a [`set`](https://docs.python.org/3/tutorial/datastructures.html#sets) would be faster because you've amortized the expensive point-in-poly operation, but this only starts to make sense when your `stations_df` array is significantly larger than the number of cells you generate for all of the polygons (or you've converted the polygons once and stored them in a database so the "live" execution path for the user is faster at the expense of a slower offline path).
